### ISSUE TITLE
Move model instantiation to Lightning module

### ISSUE THREAD

**Issue by gorold** [awslabs/gluon-ts#2661]
## Description
(A clear and concise description of what the feature is.)
Model instantiation should be moved to the Lightning module instead. This is because LightningModule saves all the input arguments to ```__init__``` when ```save_hyperparameters()``` , into a property ```self.hparams```. This includes the whole model when the model is fed as an input. Furthermore, these are stored into ```hparams.yaml```, and we are essentially saving the model weights into this file. When model size grows larger, this is takes up quite a bit of additional space.

```python
def create_lightning_module(self) -> DeepARLightningModule:
    return DeepARLightningModule(
        freq, context, ...
     )


class DeepARLightningModule(pl.LightningModule)
    def __init__(
        self,
        freq,
        context,
        ...
    )
    super().__init__()
    self.save_hyperparameters()
    self.model = DeepARModel(...)
```

## References
- list reference and related literature
- list known implementations

**Comment by lostella** [comment]
I’m wondering if `*args, **kwargs` can be used here, since otherwise there will be a lot of annoying duplication between the LightningModule and the Module?

**Comment by jaheba** [comment]
I guess the `Estimator` does not need to know about the `Model`, only about the `LightningModule`.

What's a bit annoying is that we would track the model parameters on all three classes.

**Comment by jaheba** [comment]
Fixed in #2688
